go-controller/pkg/ovn/gateway.go

```diff
@@ -383,43 +367,47 @@ func (gw *GatewayManager) GatewayInit(
 			types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
 			types.TopologyExternalID: gw.netInfo.TopologyType(),
 		}
-		if gw.netInfo.TopologyType() == types.Layer2Topology {
-			node, err := gw.watchFactory.GetNode(nodeName)
-			if err != nil {
-				return fmt.Errorf("failed to fetch node %s from watch factory %w", node, err)
-			}
-			tunnelID, err := util.ParseUDNLayer2NodeGRLRPTunnelIDs(node, gw.netInfo.GetNetworkName())
-			if err != nil {
-				if util.IsAnnotationNotSetError(err) {
-					// remote node may not have the annotation yet, suppress it
-					return types.NewSuppressedError(err)
-				}
-				// Don't consider this node as cluster-manager has not allocated node id yet.
-				return fmt.Errorf("failed to fetch tunnelID annotation from the node %s for network %s, err: %w",
-					nodeName, gw.netInfo.GetNetworkName(), err)
+	}
+	if gw.netInfo.TopologyType() == types.Layer2Topology {
```

Comment:
why break this out from checking if the network is secondary or not? How did this ever work without setting it also for primary network?

Reply:
because these ifs do logically different things, and why would you have 2 levels of nesting when 1 works

Reply:
ok I think I was confused that IsSecondary means "is secondary network" when really it means "is not default network". Something else we need to clean up. The code is OK here since CDN is never layer 2.